Feature payments

Mishgan55 opened the pull request.

Mishgan55 committed:
1.Created entity Booking
2.Added method for Car for finding all available cars
3.Refactored DB

Mishgan55 committed:
1. Created entity Payment
2. Now user can pay for booking by Stripe

MShablinskaya commented on src/main/java/innowise/khorsun/carorderservice/service/impl/StripeServiceImpl.java:
```diff
@@ -0,0 +1,101 @@
+    }
+
+    @Override
+    public PaymentDto getPaymentFromSession(SessionCreateParams params, PaymentRequestModel paymentRequestInfoDto) {
```
можно зарефакторить, из агрументов сессию убрать, а использовать метод создающий сессию

Mishgan55 committed:
1.Refactored UserServiceImpl
2. Refactored method createStripeSession in PaymentController

MShablinskaya commented on src/main/java/innowise/khorsun/carorderservice/service/PaymentService.java:
```diff
@@ -0,0 +1,17 @@
+
+@Service
+public interface PaymentService {
+    Payment save(Payment payment);
```
В сервисах мы работаем с дто, все таки.

MShablinskaya commented on src/main/java/innowise/khorsun/carorderservice/service/impl/PaymentServiceImpl.java:
```diff
@@ -0,0 +1,83 @@
+
+    @Override
+    public BigDecimal calculatePaymentAmount(Integer userId) {
+        Booking booking = bookingRepository
```
а если у пользователся несоклько заказов в таком статусе?

MShablinskaya replied:
Какой подход решения выбрал?

Mishgan55 replied:
В BookingServiceImpl есть проверка в методе checkIfActiveBookingExists()

MShablinskaya commented on src/main/java/innowise/khorsun/carorderservice/service/impl/PaymentServiceImpl.java:
```diff
@@ -0,0 +1,83 @@
+public class PaymentServiceImpl implements PaymentService {
+
+    private final PaymentRepository paymentRepository;
+    private final BookingRepository bookingRepository;
```
в сервисе мы используем только одноименные репозитории. Все остальное - только сервисами.

Mishgan55 committed:
1.Refactored PaymentService
2.Refactored booking adding logic

MShablinskaya commented on src/main/java/innowise/khorsun/carorderservice/service/impl/BookingServiceImpl.java:
```diff
@@ -0,0 +1,88 @@
+    private final BookingRepository bookingRepository;
+    private final BookingMapper bookingMapper;
+
+    private final CarRepository carRepository;
```
об использовании "чужих" репо уже был разговор

MShablinskaya commented on src/main/java/innowise/khorsun/carorderservice/entity/Payment.java:
```diff
@@ -0,0 +1,46 @@
+    private BigDecimal paymentAmount;
+    @Column(name = "type")
+    @Enumerated(value = EnumType.STRING)
+    private Type type;
```
как отслеживаешь дату оплаты?

MShablinskaya replied:
как отслеживаешь дату оплаты?

Mishgan55 committed:
1.Created tests for UserServiceImpl
2.Created tests for CarServiceImpl
3.Started testing CarServiceImpl

Mishgan55 committed:
1.Add column payment_date to payments table
2.Remade all strings using ResourceBundle and created messages.properties
3.Added a check that the user has already returned the order
4.Refactored all Services

Mishgan55 committed:
1.Add column payment_date to payments table
2.Remade all strings using ResourceBundle and created messages.properties
3.Added a check that the user has already returned the order
4.Refactored all Services

Mishgan55 committed:
Merge remote-tracking branch 'origin/feature-payments' into feature-payments

Mishgan55 committed:
Create PropertyUtil class and refactored services

Mishgan55 committed:
Some refactoring and create method for get bookings by userID

MShablinskaya commented on src/main/java/innowise/khorsun/carorderservice/service/impl/CarServiceImpl.java:
```diff
@@ -68,13 +79,31 @@ public void editCar(Integer id, CarUpdateDto carDto) {
+        car.setIsAvailable(status);
+        carRepository.save(car);
+    }
 }
```
Объединить эти два метода в 1, унифицировать

Mishgan55 committed:
Refactoring method changeCarAvailability
